fil-proofs-tooling/src/bin/benchy/rational_post.rs

```diff
@@ -0,0 +1,167 @@
+use std::collections::BTreeMap;
+use std::fs::{create_dir_all, OpenOptions};
+use std::io::stdout;
+use std::time::{Duration, Instant};
+
+use cpu_time::ProcessTime;
+use fil_proofs_tooling::metadata::Metadata;
+use filecoin_proofs::fr32::write_padded;
+use filecoin_proofs::pieces::get_aligned_source;
+use filecoin_proofs::types::{
+    PaddedBytesAmount, PoRepConfig, PoRepProofPartitions, PoStConfig, SectorSize,
+    UnpaddedBytesAmount,
+};
+use filecoin_proofs::{generate_post, seal, verify_post, PrivateReplicaInfo, PublicReplicaInfo};
+use log::info;
+use rand::random;
+use storage_proofs::sector::SectorId;
+
+// All files opened and written to will have this file path preface.
+const BASE_DIR: &str = "/var/tmp/benchy-rational-post";
+
+// The seed for the rng used to generate which sectors to challenge.
+const CHALLENGE_SEED: [u8; 32] = [0; 32];
+
+const PROVER_ID: [u8; 31] = [0; 31];
+const SECTOR_ID: u64 = 0;
+const N_PARTITIONS: PoRepProofPartitions = PoRepProofPartitions(1);
+
+#[derive(Serialize)]
+#[serde(rename_all = "kebab-case")]
+struct Inputs {
+    sector_size: usize,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "kebab-case")]
+struct Outputs {
+    proving_cpu_time_ms: u64,
+    proving_wall_time_ms: u64,
+    verifying_wall_time_ms: u64,
+    verifying_cpu_time_ms: u64,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "kebab-case")]
+struct Report {
+    inputs: Inputs,
+    outputs: Outputs,
+}
+
+impl Report {
+    /// Print all results to stdout
+    pub fn print(&self) {
+        let wrapped = Metadata::wrap(&self).expect("failed to retrieve metadata");
+        serde_json::to_writer(stdout(), &wrapped).expect("cannot write report JSON to stdout");
+    }
+}
+
+struct FuncMeasurement<T> {
+    cpu_time: Duration,
+    wall_time: Duration,
+    return_value: T,
+}
+
+fn measure<T, F>(f: F) -> Result<FuncMeasurement<T>, failure::Error>
+where
+    F: FnOnce() -> Result<T, failure::Error>,
+{
+    let cpu_time_start = ProcessTime::now();
+    let wall_start_time = Instant::now();
+
+    let x = f()?;
+
+    Ok(FuncMeasurement {
+        cpu_time: cpu_time_start.elapsed(),
+        wall_time: wall_start_time.elapsed(),
+        return_value: x,
+    })
+}
+
+pub fn run(sector_size: usize) -> Result<(), failure::Error> {
+    info!("Benchy Rational PoST: sector-size={}", sector_size);
+
+    let sector_size_unpadded_bytes_ammount =
+        UnpaddedBytesAmount::from(PaddedBytesAmount(sector_size as u64));
+
+    // Create a directory and setup some files, we will create one file for each: the staged sector,
+    // sealed sector, and the sealed sector's comm_r. `ss` in the filenames stands for
+    // "sector-size".
+    create_dir_all(BASE_DIR)?;
```

Comment:
BLOCKING: This implementation assumes that there will be only one Rational PoSt benchmark running on the same system at any point in time - and if of with-lock.sh is used as intended (and has no bugs), that assumption will be true. I don't think there's a good reason to encode that assumption in this function, though.

Fortunately, removing that assumption is as simple as replacing your staged_path (and sealed_path) with something like:

```
use tempfile::NamedTempFile;
let mut staged_file = NamedTempFile::new().expect("could not create named temp file");
let staged_path = staged_file.path();
```

This will help to ensure that each `run` call is atomic.